nanoFramework.Hardware.Esp32.Rmt/ReceiverChannelSettings.cs

```diff
@@ -0,0 +1,109 @@
+﻿//
+// Copyright (c) .NET Foundation and Contributors
+// See LICENSE file in the project root for full license information.
+//
+
+using System;
+
+namespace nanoFramework.Hardware.Esp32.Rmt
+{
+    /// <summary>
+    /// <see cref="ReceiverChannel"/>'s settings class.
+    /// </summary>
+    /// <remarks>
+    /// All changes made to properties of this class are ignored after the <see cref="ReceiverChannel"/> is initialized.
+    /// The equivalent properties in the channel instance can be used to make on-the-fly changes to the configurations.
+    /// </remarks>
+    public sealed class ReceiverChannelSettings : RmtChannelSettings
+    {
+        private ushort _idleThreshold;
+        private bool _enableFilter;
+        private byte _filterThreshold;
+        private TimeSpan _receiveTimeout;
+
+        /// <summary>
+        /// Gets or sets the idle threshold after which the receiver will go into idle mode 
+        /// and <see cref="RmtCommand"/>s are copied into the ring buffer and availble to your code.
+        /// </summary>
+        /// <remarks>
+        /// The receive process finishes(goes idle) when no edges have been detected for Idle Threshold clock cycles.
+        /// Supported value range between 1 and 65535 (0xFFFF)
+        /// </remarks>
+        public ushort IdleThreshold
+        {
+            get => _idleThreshold;
+            set
+            {
+                if (value == 0)
+                {
+                    throw new ArgumentOutOfRangeException();
+                }
+
+                _idleThreshold = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the filter state. 
+        /// If enabled, the receiver will ignore pulses with widths less than specified in <see cref="FilterThreshold"/>.
+        /// </summary>
+        public bool EnableFilter 
+        { 
+            get => _enableFilter; 
+            set => _enableFilter = value;
+        }
+
+        /// <summary>
+        /// Gets or sets the threshold for pulse widths to ignore when <see cref="EnableFilter"/> is set to <see langword="true"/>.
+        /// </summary>
+        public byte FilterThreshold 
+        { 
+            get => _filterThreshold; 
+            set => _filterThreshold = value;
```

Comment:
any minimum/maximum value or 0 to 255 is all acceptable?

Reply:
None. According to IDF 4.4.3:
"Enable a filter on the input of the RMT receiver - filter_en.
A threshold of the filter, set in the number of ticks - filter_ticks_thresh. Pulses shorter than this setting will be filtered out. Note, that the range of entered tick values is [0..255]."